### PR TITLE
man.vim: Fixes

### LINKED ISSUE #6828: man.vim: error `b:man_sect` undefined on `set ft=man` when buffer is `set nomodifiable`
```
$ nvim --version
NVIM v0.2.1-258-g1b7a9bf4d
Build type: RelWithDebInfo
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fstack-protector-strong -Wconversion -DNVIM_MSGPACK_HAS_FLOAT32 -O2 -g -DDISABLE_LOG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/home/olmo/.cache/pacaur/neovim-git/src/neovim-git/build/config -I/home/olmo/.cache/pacaur/neovim-git/src/neovim-git/src -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/usr/include -I/home/olmo/.cache/pacaur/neovim-git/src/neovim-git/build/src/nvim/auto -I/home/olmo/.cache/pacaur/neovim-git/src/neovim-git/build/include
Compiled by olmo@arch-desk

Optional features included (+) or not (-): +acl   +iconv    +jemalloc +tui      
For differences from Vim, see :help vim-differences

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"
```
- Vim (version: latest) behaves differently? Yes.
- Operating system/version:
```
$ uname -a
Linux arch-desk 4.11.3-1-ARCH #1 SMP PREEMPT Sun May 28 10:40:17 CEST 2017 x86_64 GNU/Linux
```
- Terminal name/version:
```
$ urxvt --version
urxvt: "version": unknown or malformed option.
rxvt-unicode (urxvt) v9.22 - released: 2016-01-23
options: perl,xft,styles,combining,blink,iso14755,encodings=eu+vn+jp+jp-ext+kr+zh+zh-ext,fade,transparent,tint,pixbuf,XIM,frills,selectionscrolling,wheel,slipwheel,smart-resize,cursorBlink,pointerBlank,scrollbars=plain+rxvt+NeXT+xterm
$ echo $TERM
rxvt-unicode-256color
```

### Actual behaviour
I used [ViewDoc](https://github.com/powerman/vim-plugin-viewdoc/blob/master/plugin/viewdoc_man.vim) to view manpages in Vim, but when I recently switched to NeoVim I started getting the following error when I open any man page:

```
Error detected while processing function man#init_pager:
line    2:
E21: Cannot make changes, 'modifiable' is off
E21: Cannot make changes, 'modifiable' is off
Error detected while processing /usr/share/nvim/runtime/syntax/man.vim:
line   29:
E121: Undefined variable: b:man_sect
E15: Invalid expression: b:man_sect =~# '^[23]'
Press ENTER or type command to continue
```

Then when I press any key to continue the correct man page does show up.

From the error above I thought I'd comment the line of the ViewDoc plugin that changes the filetype of the buffer, so that the `man` syntax file wasn't executed, and indeed that solved my problems.

So I went looking in `$VIMRUNTIME/syntax/man.vim` and found
```vim
if !exists('b:man_sect')
  call man#init_pager()
endif
if b:man_sect =~# '^[23]'
  " ...
```
And in `$VIMRUNTIME/autoload/man.vim`
```vim
function! man#init_pager() abort
  execute 'silent keeppatterns keepjumps %substitute,.\b\|\e\[\d\+m,,e'.(&gdefault?'':'g')
  " ...
```
Note the `%substitute`, which requires the buffer to be modifiable (there are other modifying commands in `man#init_pager()`). However, because ViewDoc had set the buffer to be `setlocal nomodifiable` before it does `setlocal ft=man`, the substitute fails and causes `man#init_pager()` to abort *before setting* `b:man_sect` and so `b:man_sect` is still undefined after the call to `man#init_pager()` in the syntax file, and so an error is generated on the `if b:man_sect =~# '^[23]'` line...

I haven't really checked *why* things are this way, but I find it really strange that a *syntax* depends on the *modifiability* of a buffer... What are your views on this?

### Expected behaviour

No errors when setting any syntax, just because the file is `set nomodifiable`.

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:set nomodifiable
:set ft=man
```

### LINKED ISSUE #11435: man.vim: add an option to disable bold highlighting
I prefer no bold highlighting in my manpages.

I can easily disable with:

```
hi manBold ctermfg=NONE ctermbg=NONE cterm=NONE gui=bold
```

Which works for terminal neovim because gui=bold prevents the default manBold group from becoming active.

It'd be nice to have an option for this.

### LINKED ISSUE #11436: man.vim: use hard wrap instead soft wrap
See https://github.com/neovim/neovim/pull/9023 for background discussion.

man.vim was changed to use vim's native wrapping by default instead of man's normal wrapping 

However, this interacts poorly with the formatting of many man pages. Some previous discussion of this is at https://github.com/neovim/neovim/issues/10748 regarding tables but I find it results in poor formatting in general.

image

As you can see, the header has been poorly formatted because man.vim sets $MANWIDTH to 999.

Similar situation with the footer.

https://github.com/neovim/neovim/blob/525bb1b55d8771939f0a9d72a532abc8f2134445/runtime/autoload/man.vim#L142

We should just hard wrap by default and not support the native vim wrapping imo. There's no good way to make it fool proof.

At the very least, given the issues, it should not be the default.

There is also precedence for hard wrapping by default as that is `:help`'s behaviour.

Another reason its nice to hard wrap by default is my terminals can be wide but I rarely want my text to span the full width, it's much easier to read narrow. Vim's soft wrapping as of now always goes end to end with the window, you can't set an arbitrary width.

### PR DESCRIPTION
**Pull request by nhooyr**
Closes #11436
Closes #11435

### CODE REVIEW

File: runtime/autoload/man.vim [hunk 1]
```diff
@@ -104,6 +104,9 @@ endfunction
 
 " Run a system command and timeout after 30 seconds.
 function! s:system(cmd, ...) abort
+  " See https://github.com/neovim/neovim/issues/9271
+  unlet $MANSECT
```
Comment (justinmk) [inline runtime/autoload/man.vim]: not sure this is worth doing, see https://github.com/neovim/neovim/issues/9271#issuecomment-557951956

Problem will occur if a colon exists in MANPATH also. See also https://github.com/neovim/neovim/pull/9159#issuecomment-433698704 for details.

File: runtime/doc/filetype.txt [hunk 2]
```diff
@@ -548,8 +548,10 @@ Variables:
 *g:ft_man_folding_enable*   Fold manpages with foldmethod=indent foldnestmax=1.
 *b:man_default_sects*       Comma-separated, ordered list of preferred sections.
                           For example in C one usually wants section 3 or 2: >
-                               :let b:man_default_sections = '3,2'
-*g:man_hardwrap*            Hard-wrap to $MANWIDTH. May improve layout.
+                               let b:man_default_sections = '3,2'
```
Comment (justinmk) [inline runtime/doc/filetype.txt]: colon prefix is used to indicate vimscript.

File: runtime/doc/filetype.txt [hunk 3]
```diff
@@ -548,8 +548,10 @@ Variables:
...
+*g:man_hardwrap*            Hard-wrap to $MANWIDTH or window width if $MANWIDTH is
+                          empty. Enabled by default. Disable to enable soft
```
Comment (justinmk) [inline runtime/doc/filetype.txt]:
```suggestion
                          empty. Enabled by default. Set |TRUE| enable soft
```

File: runtime/autoload/man.vim (hunk 1 above)
Comment (nhooyr) [reply runtime/autoload/man.vim]: Removed.

**Comment by nhooyr** [comment]
Fixed.

**Pull request merged**